src/ripple/app/misc/impl/TxQ.cpp

```diff
@@ -1529,6 +1530,37 @@ TxQ::accept(Application& app, OpenView& view)
         }
     }
 
+    // All transactions that can be moved out of the queue into the open
+    // ledger have been. Rebuild the queue using the open ledger's
+    // parent hash, so that transactions paying the same fee are
+    // reordered.
+    LedgerHash const& parentHash = view.info().parentHash;
+#if !NDEBUG
+    auto const startingSize = byFee_.size();
+    assert(parentHash != parentHash_);
+    parentHash_ = parentHash;
+#endif
+    // byFee_ doesn't "own" the candidate objects inside it, so it's
+    // perfectly safe to wipe it and start over, repopulating from
+    // byAccount_.
+    //
+    // In the absence of a "re-sort the list in place" function, this
+    // was the fastest method tried to repopulate the list.
+    // Other methods included: create a new list and moving items over one at a
+    // time, create a new list and merge the old list into it.
+    byFee_.clear();
+
+    MaybeTx::parentHashComp = parentHash;
+
+    for (auto& [_, account] : byAccount_)
+    {
+        for (auto& [_, candidate] : account.transactions)
+        {
+            byFee_.insert(candidate);
+        }
+    }
+    assert(byFee_.size() == startingSize);
```

Comment:
That's a good suggestion, and I tried it out. The test code is on my https://github.com/ximinez/rippled/commits/txq-order-lcl-sorttests branch. It ended up being consistently slower than the method here. I suspect that's because of two reasons:
1. Because `byFee_` is intrusive, an item can't be in two `FeeMultiSets` at the same time, so I still have to `clear` the list, and repopulate it.
2. Now I'm basically doing the same thing as above, but with the extra overhead of creating a new object, and doing the `swap`.

Reply:
That's unfortunate. `swap` should be cheap, but oh well.